tests/unit_tests/forward_model_runner/test_job.py

```diff
@@ -67,14 +72,62 @@ def max_memory_per_subprocess_layer(layers: int) -> int:
         max_seen = 0
         for status in job.run():
             if isinstance(status, Running):
-                max_seen = max(max_seen, status.max_memory_usage)
+                max_seen = max(max_seen, status.memory_status.max_rss)
         return max_seen
 
     max_seens = [max_memory_per_subprocess_layer(layers) for layers in range(3)]
     assert max_seens[0] < max_seens[1]
     assert max_seens[1] < max_seens[2]
 
 
+@pytest.mark.usefixtures("use_tmpdir")
+def test_memory_profile_in_running_events():
+    scriptname = "increasing_memory.py"
+    with open(scriptname, "w", encoding="utf-8") as script:
+        script.write(
+            textwrap.dedent(
+                """\
+            #!/usr/bin/env python
+            import time
+            somelist = []
+
+            for _ in range(20):
+                # 1 Mb allocated pr iteration
+                somelist.append(b' ' * 1024 * 1024)
+                time.sleep(0.02)"""
+            )
+        )
+    executable = os.path.realpath(scriptname)
+    os.chmod(scriptname, stat.S_IRWXU | stat.S_IRWXO | stat.S_IRWXG)
+
+    fm_step = Job(
+        {
+            "executable": executable,
+            "argList": [""],
+        },
+        0,
+    )
+    fm_step.MEMORY_POLL_PERIOD = 0.01
+    emitted_rss_values = []
+    for status in fm_step.run():
+        if isinstance(status, Running):
+            emitted_rss_values.append(status.memory_status.rss)
+
+    # Any asserts on the emitted_rss_values easily becomes flaky, so be mild:
+
+    assert (
+        np.diff(np.array(emitted_rss_values[:-3]))
+        > 0
+        # Avoid the tail of the array, then the process is tearing down
+    ).all(), f"Emitted memory usage not increasing, got {emitted_rss_values[:-3]=}"
+
+    assert (
```

Comment:
do you want to (maybe) include that the timestamps are in a certain interval?

Reply:
done.